react/src/components/DeleteMapModal/DeleteMapModal.tsx

```diff
@@ -0,0 +1,77 @@
+import React, { useState } from 'react';
+import { Modal, ModalHeader, ModalBody, ModalFooter } from 'reactstrap';
+import { Button, SectionMessage } from '../../core-components';
+import styles from './DeleteMapModal.module.css';
+import { Project } from '../../types';
+import { useDeleteProject } from '../../hooks/projects/';
+
+type DeleteMapModalProps = {
+  isOpen: boolean;
+  toggle: () => void;
+  projectId?: number;
+  project?: Project;
+};
+
+const DeleteMapModal = ({
+  isOpen,
+  toggle: parentToggle,
+  projectId,
+  project,
+}: DeleteMapModalProps) => {
+  const [errorMessage, setErrorMessage] = useState('');
+  const { mutate: deleteProject, isLoading: isDeletingProject } =
+    useDeleteProject(projectId);
+  const handleClose = () => {
+    setErrorMessage(''); // Clear the error message
+    parentToggle(); // Call the original toggle function passed as a prop
+  };
+
+  const handleDeleteProject = () => {
+    deleteProject(undefined, {
+      onSuccess: () => {
+        handleClose();
+      },
+      onError: () => {
+        setErrorMessage('There was an error deleting your project.');
+      },
+    });
+  };
+
+  return (
+    <Modal isOpen={isOpen} toggle={handleClose} className={styles.root}>
+      <ModalHeader toggle={handleClose}>
+        Delete Map: {project?.name}{' '}
+      </ModalHeader>
+      <ModalBody>
+        {project?.deletable
+          ? 'Are you sure you want to delete this map? All associated features, metadata, and saved files will be deleted. THIS CANNOT BE UNDONE.'
+          : "This map is not able to be deleted either because the map is public or because you don't have permission."}
```

Comment:
I was thinking for UI reasons, opening the modal and telling them why they can't delete it would make it more clear for the user rather than just having disable buttons with no mesage? I can keep it as it works now though if that makes more sense for this part of the redesign

Reply:
Also, doesn't a public map make it unable to be deleted? Maybe I have the logic flow wrong.

Reply:
I cleared this up. It should only not be deleted if a user doesn't have permission as an admin or creator. Will update the display message. I can still just disable this trash icon if that's preferred.